.github/workflows/ci.yaml

```diff
@@ -217,35 +200,29 @@ jobs:
       # Ignore the failure of a step and avoid terminating the job.
       continue-on-error: true
       uses: actions/cache@v3
+      id: deps_cache
       with:
         # This path is specific to Ubuntu
         path: ${{ github.workspace }}/cget
         # Look to see if there is a cache hit for the corresponding requirements file
         key: 
           ${{ matrix.os }}-cget-4-${{ hashFiles('requirements.txt', 'dev-requirements.txt') }}
+        restore-keys: |
           ${{ matrix.os }}-cget-4-
 
 
     - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
-        rbuild prepare -d cget -s gh
-        sudo apt-get install -y lld
-    - name: Prepare timestamp
-      id: cache_timestamp
-      shell: bash 
-      run: echo timestamp="$(date +'%Y-%m-%dT%H:%M:%S')" >> $GITHUB_OUTPUT
+      if: steps.deps_cache.outputs.cache-hit != 'true'
+      run: rbuild prepare -d cget -s gh
 
     - name: Cache files for ccache
       # Ignore the failure of a step and avoid terminating the job.
       continue-on-error: true
-      uses: pat-s/always-upload-cache@v2.1.3
+      uses: actions/cache@v3 
```

Comment:
Will `actions/cache@v3` upload the cache even when there is a build failure?

Reply:
From the issue [here](https://github.com/actions/cache/issues/92#issuecomment-1362456629) we need to use action/cache/save. This section describes exactly the same scenario of saving the cache even when there is a test failure:

https://github.com/actions/cache/tree/main/save#always-save-cache